### ISSUE TITLE
Nginx plugin Phalcon 4.x compatibility 

### ISSUE THREAD

**Issue by AdSchellevis** [opnsense/plugins#2343]
**Important notices**
Before you add a new report, we ask you kindly to acknowledge the following:

[x] I have read the contributing guide lines at https://github.com/opnsense/plugins/blob/master/CONTRIBUTING.md

[x] I have searched the existing issues and I'm convinced that mine is new.

[x] The title contains the plugin to which this issue belongs

**Describe the bug**
In https://github.com/opnsense/core/issues/4012 we upgraded our Phalcon version to 4.1 and refactored the relevant code, it seems that the Nginx plugin contains some custom constraints which depend on Phalcon and might be subjected to change.

The master branch and upcoming development release of OPNsense will contain all required changes in core, the Nginx plugin needs to be validated and altered if needed.

```
# grep -r Phalcon www/nginx/
www/nginx/src/opnsense/mvc/app/controllers/OPNsense/Nginx/Api/SettingsController.php:     * @throws \Phalcon\Validation\Exception on validation errors
www/nginx/src/opnsense/mvc/app/controllers/OPNsense/Nginx/Api/SettingsController.php:     * @throws \Phalcon\Validation\Exception on validation errors
www/nginx/src/opnsense/mvc/app/controllers/OPNsense/Nginx/Api/SettingsController.php:     * @throws \Phalcon\Validation\Exception
www/nginx/src/opnsense/mvc/app/models/OPNsense/Base/Constraints/NgxBusyBufferConstraint.php:use Phalcon\Validation\Message;
www/nginx/src/opnsense/mvc/app/models/OPNsense/Base/Constraints/NgxBusyBufferConstraint.php:    public function validate(\Phalcon\Validation $validator, $attribute)
www/nginx/src/opnsense/mvc/app/models/OPNsense/Base/Constraints/NaxsiIdentifierConstraint.php:use Phalcon\Validation\Message;
www/nginx/src/opnsense/mvc/app/models/OPNsense/Base/Constraints/NaxsiIdentifierConstraint.php:    public function validate(\Phalcon\Validation $validator, $attribute)
www/nginx/src/opnsense/service/templates/OPNsense/Nginx/webgui.conf:  # Phalcon ui and api routing
```

**Comment by AdSchellevis** [comment]
@fabianfrz do you have time to handle this one?

**Comment by fabianfrz** [comment]
@AdSchellevis Is there a way I can test that? So that I do have a phalcon 4.1 on a VM to test with?

**Comment by AdSchellevis** [comment]
@fabianfrz the new phalcon version is shipped as a development package in 21.1.5, so switching a vm to development should be enough to test your changes. I haven't looked at ui_devtools yet, still need to recompile some stuff on my mac

**Comment by fabianfrz** [comment]
@AdSchellevis I will try to have a look at it this weekend then.
